src/main/kotlin/uk/co/poolefoundries/baldinggate/core/BaldingGateGame.kt

```diff
@@ -230,14 +198,14 @@ class BaldingGateGame : Game() {
     private fun playerMove(target: PositionComponent) {
         val distance = selectedEntity.position!!.manhattanDistance(target)
         var tempPos = selectedEntity.position!!
-        val positions = Array<PositionComponent>()
+        val positions = mutableListOf<PositionComponent>()
         positions.add(tempPos)
         // TODO: pathfinding
         for (step in 0 until minOf(selectedEntity.speed, distance)) {
             tempPos += tempPos.direction(target)
             positions.add(tempPos)
         }
-        pendingAnimations.add(selectedEntity.entity?.let { Animation(it, positions) })
+        selectedEntity.entity?.let { engine.getSystem(AnimationSystem::class.java).addAnimation(it, positions) }
```

Comment:
It's a bit weird having SelectedEntity have an entity field that could be null. I think I'd rather have a "PlayerTurnSystem.getSelectedEntity()" that might return null. I guess this can be cleaned up later when we come round to the player turn system.